internal/precache/workload/pullImages.go

```diff
@@ -0,0 +1,200 @@
+/*
+ * Copyright 2023 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this inputFilePath except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package workload
+
+import (
+	"os"
+	"os/exec"
+	"strconv"
+	"sync"
+
+	"github.com/openshift-kni/lifecycle-agent/internal/precache"
+	log "github.com/sirupsen/logrus"
+)
+
+// MaxRetries is the max number of retries for pulling an image before marking it as failed
+const MaxRetries int = 5
+
+// Podman auth-file related constants
+const (
+	EnvAuthFile     string = "PULL_SECRET_PATH"
+	DefaultAuthFile string = "/var/lib/kubelet/config.json"
+)
+
+// executeCmd execute shell commands
+func executeCmd(suppressCmdOutput bool, cmd string, args ...string) (err error) {
+
+	logger := log.StandardLogger()
+	logWriter := logger.Writer()
+
+	log.Debugf("Executing %s with args %s", cmd, args)
+	execCmd := exec.Command(cmd, args...)
+
+	if suppressCmdOutput {
+		execCmd.Stdout = nil
+		execCmd.Stderr = nil
+	} else {
+		execCmd.Stdout = logWriter
+		execCmd.Stderr = logWriter
+	}
+
+	if err = execCmd.Run(); err != nil {
+		if !suppressCmdOutput {
+			log.Error(err)
+		}
+	}
+
+	_ = logWriter.Close()
+	return err
+}
+
+// CheckPodman verifies that podman is running by checking the version of podman
+func CheckPodman() bool {
+	if err := executeCmd(false, "podman", []string{"version"}...); err != nil {
+		return false
+	}
+	return true
+}
+
+// podmanImgExists reports the existence of the given image via podman
+func podmanImgExists(image string) bool {
+	if err := executeCmd(true, "podman", []string{"image", "exists", image}...); err != nil {
+		return false
+	}
+	return true
+}
+
+// podmanImgPull pulls the image via podman
+func podmanImgPull(image, authFile string) error {
+	args := []string{"pull", image}
+	if authFile != "" {
+		args = append(args, []string{"--authfile", authFile}...)
+	}
+	return executeCmd(false, "podman", args...)
+}
+
+// pullImage attempts to pull an image using the Podman API
```

Comment:
nit: Should probably remove comment references to Podman API

Reply:
Changed from API -> CLI